### ISSUE TITLE
Semantic Segmentation of High Resolution Imagery

### ISSUE THREAD

**Issue by mortcanty** [google/earthengine-community#453]
**What is the objective of the proposed tutorial?**
Semantic segmentation is of course a special application. But I hope the tutorial will give the user a feeling for the many exciting possibilities for geospatial research offered by combining Earth Engine functionality with machine learning within a Python/GEE/Colab/TensorFlow environment.

**What is the scope of the proposed tutorial?**
Fully convolutional neural networks (FCNs) are commonly used for semantic image segmentation, essentially the assignment of every pixel in an image to one of two or more categories. In this tutorial I describe in detail a popular FCN architecture, called UNet, to perform urban building recognition: the identification within an arbitrarily complex remote sensing image of houses, schools, commercial edifices, etc. The model is trained on the four-band NAIP imagery in the GEE archive, making use of the semantic labels dataset for the [Inria Aerial Imaging Labeling Benchmark](https://hal.inria.fr/hal-01468452/document), published in 2016. The tutorial illustrates model creation with TensorFlow, setting up a data pipeline, model training, evaluation and inference over new data.

**Please provide an outline of the structure of the proposed tutorial?**
Section 1
  The training labels (The Inria competition)
Section 2
  Programming a fully convolutional network (UNet) wth Tensorflow
Section 3
  Construction of a training set from GEE NAIP imagery
  Setting up a data pipeline for training and testing
Section 4.
  Model complilation and training on a Colab GPU runtime
Section 5.
  Model evaluation (visual assessment, examination of low level filters, binary accuracy and intersection over union metrics)
Section 6.
  Inference: Segmentation of arbitrary scenes on the NAIP archive. 
Section 7.
  Outlook: Available geospatial training data and other machine learning applications.

**In what format will you be submitting the tutorial?**
Colab including iPyLeaflet interactive maps. A draft Colab notebook is available [here](https://drive.google.com/file/d/1yL-V3q2a9VONiukcWHmW010v7A_2aRxA/view?usp=sharing)

This request will be reviewed by the Earth Engine community maintainers, who
will reply on this issue tracker with any questions or suggestions. Once
approved, this issue will be assigned to you and you can begin work on the
tutorial following instructions in
[Writing a tutorial](https://developers.google.com/earth-engine/tutorials/community/write).
When creating your Pull Request, enter "Closes #issueno" in the description of
your Pull Request to link the tutorial to this issue.

**Comment by jdbcode** [comment]
Hi Mort!
I've scheduled time 2022-02-08 to review the notebook and see if there is anything that should be addressed before you submit a pull request.
I don't have access to the Colab notebook you linked to (I requested it in the popup).

**Comment by mortcanty** [comment]
Great! I've sent you the link again directly from the  Colab share menu. You've probaly received it twice now :)

**Comment by jdbcode** [comment]
Confirming I can see it in Colab now; thanks for the share!

**Comment by mortcanty** [comment]
Hi Justin,
Serious reservations? Or haven't you gotten round to it yet? 
Thanks

**Comment by mortcanty** [comment]
Thorough update, many thanks!

**Comment by jdbcode** [comment]
Mort, I'm sorry to say that we feel like this tutorial is a bad fit for publishing under the [Community Tutorial section](https://developers.google.com/earth-engine/tutorials/community/explore) of the Developer's Guide. Mostly that it is a challenge for our publishing infrastructure and has high maintenance obligations.

However, we like the content and would like it to be discoverable. Our recommendation is to add it as a Python API tutorial in the "[Awesome-GEE](https://github.com/giswqs/Awesome-GEE)" resource list (like [here](
https://github.com/giswqs/Awesome-GEE/blob/061af52ef52d78e8f26c601431e3a4321fa6e596/readme.md?plain=1#L148
)) on GitHub. If you'd like assistance adding the Colab notebook to the list, I be happy to setup the PR, just let me know.

**Comment by mortcanty** [comment]
Was kind of expecting that, and I appreciate the consideration you've given to the proposal. I'll have a look at your suggestion and get back to you if I need help. Cheers. Mort

**Comment by jdbcode** [comment]
Thanks for understanding :)
I'll close this for now, ping the thread later if you need to.

**Comment by mortcanty** [comment]
Mailed this a few days ago::

Justin, can I take you up on your offer? The Colab notebook is a public GitHub Gist:
[Semantic segmentation](https://gist.github.com/mortcanty/ac4c48e3d10e89676b7fe9b3a6f1ba3a)
The users can get the train/test dataset and a pre-trained model from my Drive (about 3GB altogether), and that's all they need.
I've juiced it up with a widget interface similar to the one I used for the S1 Tutorial series. The trained UNet model really works beautifully. Shame it doesn't fit into the Tutorials.  Anyway, I assume the PR to the "Awesome" repo just involves adding a link to my Gist, but the "contributing" section is not helpful.

Thanks again (and in advance) for you help.
